packages/shared/src/db/queries.ts

```diff
@@ -1205,6 +1208,39 @@ export const setJoinedGroupChannels = createWriteQuery(
   ['channels']
 );
 
+export const addJoinedGroupChannel = createWriteQuery(
```

Comment:
Could this be combined with `setJoinedGroupChannels`?

Reply:
This handles the specific case where we've just gotten an update about a particular channel being added, so we add to the 'default' nav section before running the update for the single channel.

I looked at using setJoinedGroupChannels, but the logic of the query doesn't make a lot of sense to me for this use case. We update all channels and set currentUserIsMember on each one based on whether the channel ID for each row is in the list of IDs we've just passed to the query. This makes sense for the use case of updating channel membership based on unreads, but for a one off channel add this will mark every other channel (other than the new channel) the user has as unjoined.

We could make setJoinedGroupChannels more complex, but I feel like two simple queries here is better than one complex one.